consent/src/main/java/com/seaplain/android/consent/PermissionRequest.java

```diff
@@ -20,14 +22,19 @@
         void onExplanationCompleted();
     }
 
-    private Activity mContext;
+    private HostWrapper mHostWrapper;
     private String[] mRequestedPermissions;
     private String[] mPermissionsThatNeedExplanation; // Cached for performance reasons
     private String[] mUnprovidedPermissions; // Cached for performance reasons
     private ExplanationListener mExplanationListener;
 
-    public PermissionRequest(@NonNull Activity context, @NonNull String... permissions) {
-        mContext = context;
+    public PermissionRequest(@NonNull Activity activity, @NonNull String... permissions) {
+        mHostWrapper = new HostWrapper(activity);
+        mRequestedPermissions = permissions;
+    }
+
+    public PermissionRequest(@NonNull Fragment fragment, @NonNull String... permissions) {
+        mHostWrapper = new HostWrapper(fragment);
```

Comment:
I might be missing something here, is there a reason this couldn't simply be `mContext = fragment.getActivity();`?

Reply:
The problem is that you'll then have to put `Consent.handle(requestCode, permissions, grantResults);` inside the activity/activities hosting your fragment. This isn't typically a problem if you put it in your `BaseActivity` but if you have a transitive dependency on `Consent` through a library, it no longer is transparent for you that the library uses `Consent` under the hood.

In `MySupportFragment` in the app inside `onRequestPermissionsResult(...)` I'd like to call: `someLibrary.handle(requestCode, permissions, grantResults);` which internally happens to call `Consent.handle(requestCode, permissions, grantResults);`.

Though, seeing how this works now with the callbacks, passing a fragment could cause leaks. Perhaps `someLibrary` can instead make a static method wrapping the `Consent` call and suggest consumers to call that in their `BaseActivity`. I'll check on it next week, but if that is fine we don't need the change. Still, it would be nice to to utilise the fact that support fragments and fragment activity already does all this delegation for us.

Reply:
Good point! I do agree that putting onus on a wrapping library is undesirable. I'll take a look at that leak concern but conceptually I don't have an issue with this change.